From cf4a16b5557be5fb3568c1de0d6cc0a18291afc9 Mon Sep 17 00:00:00 2001
From: Sergei Golubchik <serg@mariadb.org>
Date: Thu, 4 May 2023 16:05:08 +0200
Subject: [PATCH] MDEV-31057 rocksdb does not compile with gcc-13

RocksDB (in a submodule) has to include <cstdint> to use uint64_t
but it doesn't. Until the submodule is upgraded, let's replace
problematic types with something that's available
---
 storage/rocksdb/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/storage/rocksdb/CMakeLists.txt b/storage/rocksdb/CMakeLists.txt
index 15fc4dc735a8a..71259703e5a6f 100644
--- a/storage/rocksdb/CMakeLists.txt
+++ b/storage/rocksdb/CMakeLists.txt
@@ -30,6 +30,11 @@ IF(WITH_VALGRIND)
   ADD_DEFINITIONS(-DROCKSDB_VALGRIND_RUN=1)
 ENDIF()
 
+ADD_DEFINITIONS(-Duint64_t=u_int64_t)
+ADD_DEFINITIONS(-Duint32_t=u_int32_t)
+ADD_DEFINITIONS(-Duint16_t=u_int16_t)
+ADD_DEFINITIONS(-Duint8_t=u_int8_t)
+
 # We've had our builders hang during the build process. This prevents MariaRocks
 # to be built on 32 bit intel OS kernels.
 IF(CMAKE_SYSTEM_PROCESSOR MATCHES "i[36]86")
From 75bbf645a66db797be2abd3a348dce32eb753acc Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 24 Jan 2023 21:40:43 -0800
Subject: [PATCH] Add missing include <cstdio>

This is needed with GCC 13 and newer [1]

[1] https://www.gnu.org/software/gcc/gcc-13/porting_to.html

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 tpool/aio_linux.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tpool/aio_linux.cc b/tpool/aio_linux.cc
index 8a6688e6775fc..6997cbcccaba8 100644
--- a/tpool/aio_linux.cc
+++ b/tpool/aio_linux.cc
@@ -19,6 +19,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02111 - 1301 USA*/
 #ifdef LINUX_NATIVE_AIO
 # include <thread>
 # include <atomic>
+# include <cstdio>
 # include <libaio.h>
 # include <sys/syscall.h>
